### ISSUE TITLE
Clients need an API entry on OC to search for SHAREES in REMOTE trusted servers

### ISSUE THREAD

**Issue by davivel** [owncloud/core#22749]
Parting from two server with the Federation App installed and configured. Both servers trust each other, and their address books are already synchronized. In this environment, autocompletion is available in the web UI when a sharee username is partially typed, so that the search results include users in the remote and trusted server.

To implement this autocompletion in the clients, we need an entry in the OC API that returns remote users. Right now the clients use the Sharee API in OCS with this target, allowing the autocompletion of users and groups in the user's server. An easy option from our side would be that the response of this API includes remote users too.

cc @rullzer @schiesbn @PVince81 @davitol @rperezb 

cc @MTRichards @cmonteroluque 

PS: AFAIK, not urgent.

**Comment by nickvergessen** [comment]
Issue titles are not descriptions, you can do better than that :wink:

**Comment by davivel** [comment]
OK, let me try again :)

**Comment by nickvergessen** [comment]
Did you test if that is not the case already?
We check the available addressbooks, and technically the federated user lists are addressbooks

**Comment by davivel** [comment]
Yes, I checked it. From my talk with @schiesbn I understood that this was not implemented yet, not that it could be a bug.

**Comment by PVince81** [comment]
@schiesbn @rullzer @deepdiver1975 can you guys confirm ?

**Comment by DeepDiver1975** [comment]
This is already in place. Trusted servers do sync their system addressbooks which are representing the instance users. Each contacts also holds the cloud id used for remote sharing.

This was successfully tested already by qa.

**Comment by nickvergessen** [comment]
Confirmed, just set up federation again and the users are correctly listed in the dropdown.
The dropdown only asks the normal sharee endpoint, which is OCS as ready, so that should be fine.

Closing since it is already there

**Comment by davivel** [comment]
Sorry, it's not working for me.

Maybe I'm just doing the wrong query:

```curl "https://[user]:[pass]@[domain-port-path]/ocs/v2.php/apps/files_sharing/api/v1/sharees?itemType=search&page=1&perPage=50"```

It returns all the local users and groups, no remote user. The same server allow federated shares with autocompletion without problem.

**Comment by davivel** [comment]
Testing with ownCloud 9.0.0 beta 2.

In the logs of the queried server can find
```Error	OCP\Share	Sharing backend for search not found````
with every request.

**Comment by nickvergessen** [comment]
itemType share is unknown and therefor does not allow remote sharing, try `itemType=file`

**Comment by davivel** [comment]
That works.

Question: why the unknown itemType works with local users?

**Comment by nickvergessen** [comment]
because only file sharing backend knowns remote shares.
Others work for all backends

**Comment by davivel** [comment]
OK, thank you very much.
